### PR TITLE
build: maintaining dependencies

### PR DESCRIPTION
**Pull request by HyperLife1119**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Removed some useless dependencies and updated some key dependencies.

I also fixed some broken tests due to updated jasmine.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/code-editor/code-editor.spec.ts [hunk 1]
```diff
@@ -33,20 +31,20 @@ export class NzTestCodeEditorFullControlComponent {
 }
 
 describe('code editor', () => {
-  describe('basic', () => {
-    // let fixture: ComponentFixture<NzDemoCodeEditorBasicComponent>;
+  // describe('basic', () => {
```
Comment (Laffery) [inline components/code-editor/code-editor.spec.ts:34]: why skip this test? what about delete it directly?

Comment (HyperLife1119) [reply components/code-editor/code-editor.spec.ts:34]: As you can see, this is an empty test, which is not allowed in the latest jasmine.

Comment (HyperLife1119) [reply components/code-editor/code-editor.spec.ts:34]: I chose to annotate it so we can come back and refine it later. Of course, I could have just deleted it.

**Review by Laffery — CHANGES_REQUESTED** [review]
nice job!

**Review by Laffery — APPROVED** [review]

**Pull request merged**